shop/forms/checkout.py

```diff
@@ -168,9 +168,12 @@ def form_factory(cls, request, data, cart):
                 # prevent adding the same address twice
                 all_field_names = [f.name for f in cls.get_model()._meta.get_fields()]
                 filter_args = dict((attr, val) for attr, val in address_form.data.items()
-                                   if attr in all_field_names and val)
+                                   if attr in all_field_names)
```

Comment:
Alright, then we need to come up with a better fix for this. Think of this case, for example:

```
Musterfirma
Zu Hd. Herrn Hans Müller - persönlich, vertraulich!
Musterstr. 1
12345 Musterstadt
```

```
Musterfirma

Musterstr. 1
12345 Musterstadt
```

Because of the `and val`, we treat these as the same address, which is obviously not correct. I can also smell problems if the address exists with and without VAT ID, for example. That is where I ran into the problem. Unfortunately, creating a minimal reproducible case is a bit of work (my project is very different from the example project). I need to see when I find the time for it.